src/protocol/render.rs

```diff
@@ -2263,14 +2263,14 @@ pub struct ChangePictureAux {
     pub dither: Option<xproto::Atom>,
     pub componentalpha: Option<u32>,
 }
-impl Serialize for ChangePictureAux {
-    type Bytes = Vec<u8>;
-    fn serialize(&self) -> Vec<u8> {
+#[allow(dead_code, unused_variables)]
+impl ChangePictureAux {
+    fn serialize(&self, value_mask: u32) -> Vec<u8> {
         let mut result = Vec::new();
-        self.serialize_into(&mut result);
+        self.serialize_into(&mut result, value_mask);
         result
     }
-    fn serialize_into(&self, bytes: &mut Vec<u8>) {
+    fn serialize_into(&self, bytes: &mut Vec<u8>, value_mask: u32) {
         if let Some(repeat) = self.repeat {
```

Comment:
Would it make sense to `assert_eq!(value_mask, self.value_mask());`? It feels like that would get rid of most of the unused parameters.